docker: docker cannot access /etc/localtime on OS X

<!--
If your issue is a usage question, please submit it to the user mailing-list
users@riot-os.org or to the developer mailing-list devel@riot-os.org.
If your issue is related to security, please submit it to the security
mailing-list security@riot-os.org.
-->

#### Description
build with Docker on OS X fails because on OS X, /etc/localtime cannot be accessed by Docker.


#### Steps to reproduce the issue
building HelloWorld with docker:

```
$ BUILD_IN_DOCKER=1 BOARD=samr21-xpro make -C examples/hello-world/ 
Warning: no PORT set!
Launching build container using image "riot/riotbuild:latest".
docker run --rm -t -u "$(id -u)" \
	    -v '/Users/emmanuel/Documents/work/CODING/RIOT:/data/riotbuild/riotbase' \
	    -v '/Users/emmanuel/Documents/work/CODING/RIOT/cpu:/data/riotbuild/riotcpu' \
	    -v '/Users/emmanuel/Documents/work/CODING/RIOT/boards:/data/riotbuild/riotboard' \
	    -v '/Users/emmanuel/Documents/work/CODING/RIOT/makefiles:/data/riotbuild/riotmake' \
	    -v '/Users/emmanuel/Documents/work/CODING/RIOT:/data/riotbuild/riotproject' \
	    -v /etc/localtime:/etc/localtime:ro \
	    -e 'RIOTBASE=/data/riotbuild/riotbase' \
	    -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' \
	    -e 'RIOTCPU=/data/riotbuild/riotcpu' \
	    -e 'RIOTBOARD=/data/riotbuild/riotboard' \
	    -e 'RIOTMAKE=/data/riotbuild/riotmake' \
	    -e 'RIOTPROJECT=/data/riotbuild/riotproject' \
	      \
	    -e 'BOARD=samr21-xpro' \
	    -w '/data/riotbuild/riotproject/examples/hello-world/' \
	    'riot/riotbuild:latest' make  
docker: Error response from daemon: Mounts denied: 
The path /etc/localtime
is not shared from OS X and is not known to Docker.
You can configure shared paths from Docker -> Preferences... -> File Sharing.
See https://docs.docker.com/docker-for-mac/osxfs/#namespaces for more info.
.
ERRO[0000] error waiting for container: context canceled 
make: *** [..in-docker-container] Error 125
```

#### Expected results
Build succeeds.

#### Actual results
Build fails.

#### Versions
RIOT master, on OS X 10.14, with Docker 18.06.1-ce.

--- a/makefiles/docker.inc.mk
+++ b/makefiles/docker.inc.mk
@@ -94,6 +94,11 @@ _is_git_worktree = $(shell grep '^gitdir: ' $(RIOTBASE)/.git 2>/dev/null)
 GIT_WORKTREE_COMMONDIR = $(shell git rev-parse --git-common-dir)
 DOCKER_VOLUMES_AND_ENV += $(if $(_is_git_worktree),-v $(GIT_WORKTREE_COMMONDIR):$(GIT_WORKTREE_COMMONDIR))
 
+# Resolve symlink of /etc/localtime to its real path
+# This is a workaround for docker on macOS, for more information see:
+# https://github.com/docker/for-mac/issues/2396
+ETC_LOCALTIME = $(realpath /etc/localtime)
+
 # This will execute `make $(DOCKER_MAKECMDGOALS)` inside a Docker container.
 # We do not push the regular $(MAKECMDGOALS) to the container's make command in
 # order to only perform building inside the container and defer executing any
@@ -109,7 +114,7 @@ DOCKER_VOLUMES_AND_ENV += $(if $(_is_git_worktree),-v $(GIT_WORKTREE_COMMONDIR):
 	    -v '$(RIOTBOARD):$(DOCKER_BUILD_ROOT)/riotboard' \
 	    -v '$(RIOTMAKE):$(DOCKER_BUILD_ROOT)/riotmake' \
 	    -v '$(RIOTPROJECT):$(DOCKER_BUILD_ROOT)/riotproject' \
-	    -v /etc/localtime:/etc/localtime:ro \
+	    -v '$(ETC_LOCALTIME):/etc/localtime:ro' \
 	    -e 'RIOTBASE=$(DOCKER_BUILD_ROOT)/riotbase' \
 	    -e 'CCACHE_BASEDIR=$(DOCKER_BUILD_ROOT)/riotbase' \
 	    -e 'RIOTCPU=$(DOCKER_BUILD_ROOT)/riotcpu' \

